<a href="https://colab.research.google.com/github/RobsonGFerrarezi/TrabalhoAmbientesInteligentes_N2_GRUPO2/blob/main/TrabalhoN2_AmbientesInteligentes_Grupo2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

<a rel="license" href="https://faculdadesalvadorarena.org.br/"><img alt="FESA" style="border-width:0" src="https://faculdadesalvadorarena.org.br/wp-content/uploads/2022/07/logo_fesa.png" /></a><br />
**FESA - Eletiva II (2025.1)** <br/>
<br/>
**Avaliação N2** <br/>
<br/>
Aluno(a): BEATRIZ DOS SANTOS BUGLIO <br/>
Aluno(a): KAUÊ DOS SANTOS ANDRADE <br/>
Aluno(a): ROBSON GUILHERME FERRAREZI <br/>
Aluno(a): RODRIGO MORO BATALHA




---

Este notebook tem como função, análise e tratamento de dados de 4 Data Frames diferentes. Vamos ler o arquivo, ver o tipo de dados, quantidade de dados, e responder algumas perguntas.
<br/><br/>
Isso vai ser feito com a biblioteca Pandas.

In [ ]:
#importando a biblioteca:
import pandas as pd

Aqui vamos trazer os Data Frames, estamos puxando-os diretamente do GitHub, usando o link RAW para trazer a base de dados.

In [ ]:
!wget --no-check-certificate --content-disposition 'https://raw.githubusercontent.com/RobsonGFerrarezi/TrabalhoAmbientesInteligentes_N2_GRUPO2/refs/heads/main/movies.csv'
!wget --no-check-certificate --content-disposition 'https://raw.githubusercontent.com/RobsonGFerrarezi/TrabalhoAmbientesInteligentes_N2_GRUPO2/refs/heads/main/links.csv'
!wget --no-check-certificate --content-disposition 'https://raw.githubusercontent.com/RobsonGFerrarezi/TrabalhoAmbientesInteligentes_N2_GRUPO2/refs/heads/main/ratings.csv'
!wget --no-check-certificate --content-disposition 'https://raw.githubusercontent.com/RobsonGFerrarezi/TrabalhoAmbientesInteligentes_N2_GRUPO2/refs/heads/main/tags.csv'

--2025-04-04 01:08:44--  https://raw.githubusercontent.com/RobsonGFerrarezi/TrabalhoAmbientesInteligentes_N2_GRUPO2/refs/heads/main/movies.csv
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.108.133, 185.199.109.133, 185.199.110.133, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.108.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 494431 (483K) [text/plain]
Saving to: ‘movies.csv.1’

movies.csv.1        100%[===================>] 482.84K  --.-KB/s    in 0.06s   

2025-04-04 01:08:45 (7.57 MB/s) - ‘movies.csv.1’ saved [494431/494431]

--2025-04-04 01:08:45--  https://raw.githubusercontent.com/RobsonGFerrarezi/TrabalhoAmbientesInteligentes_N2_GRUPO2/refs/heads/main/links.csv
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.111.133, 185.199.108.133, 185.199.109.133, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.111.133|:443... connected.
HTT

In [ ]:
#Aqui estamos criando os DataFrames para salvar as informações.
movies = pd.read_csv('movies.csv')
links = pd.read_csv('links.csv')
ratings = pd.read_csv('ratings.csv')
tags = pd.read_csv('tags.csv')



---

Aqui vamos trabalhar com os DataFrames, um por um, verificando as informações para ver se tem algum erro, e se precisa de tratamento.




---

Movies

In [ ]:
#Verificando Linhas, Colunas
movies.shape

(9742, 3)

In [ ]:
movies.head()

movieId                               title  \
0        1                    Toy Story (1995)   
1        2                      Jumanji (1995)   
2        3             Grumpier Old Men (1995)   
3        4            Waiting to Exhale (1995)   
4        5  Father of the Bride Part II (1995)   

                                        genres  
0  Adventure|Animation|Children|Comedy|Fantasy  
1                   Adventure|Children|Fantasy  
2                               Comedy|Romance  
3                         Comedy|Drama|Romance  
4                                       Comedy

In [ ]:
movies.rename(columns={'movieId':'id_filme', 'title':'Titulo', 'genres':'Generos'}, inplace=True)

In [ ]:
movies.head()

id_filme                              Titulo  \
0         1                    Toy Story (1995)   
1         2                      Jumanji (1995)   
2         3             Grumpier Old Men (1995)   
3         4            Waiting to Exhale (1995)   
4         5  Father of the Bride Part II (1995)   

                                       Generos  
0  Adventure|Animation|Children|Comedy|Fantasy  
1                   Adventure|Children|Fantasy  
2                               Comedy|Romance  
3                         Comedy|Drama|Romance  
4                                       Comedy

In [ ]:
#iremos verificar se tem algum valor nulo, ou alguma tipagem errada.
movies.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9742 entries, 0 to 9741
Data columns (total 3 columns):
 #   Column    Non-Null Count  Dtype 
---  ------    --------------  ----- 
 0   id_filme  9742 non-null   int64 
 1   Titulo    9742 non-null   object
 2   Generos   9742 non-null   object
dtypes: int64(1), object(2)
memory usage: 228.5+ KB


Links

In [ ]:
#aqui reparem nas suas informações, e vejam se tem que renomear, ou resolver algo, façam para cada tipo
#renomear a coluna movieId	para id_filme

Tags

In [ ]:
#aqui reparem nas suas informações, e vejam se tem que renomear, ou resolver algo, façam para cada tipo
#renomear a coluna movieId	para id_filme

Ratings

In [ ]:
#aqui reparem nas suas informações, e vejam se tem que renomear, ou resolver algo, façam para cada tipo
#renomear a coluna movieId	para id_filme



---



**Questão 1 - a)** Existe alguma correlação entre o ano de lançamento do filme e sua avaliação média (média de ratings)?


Para responder essa pergunta, vou criar uma nova coluna, com nome "Ano", que indica o ano de lançamento do filme, para poder relacionar com a avaliação.

In [ ]:
#usando esse código, estaremos criando a nova coluna, e colocando o
#dado da coluna Titulo, que está entre parenteses
movies['Ano'] = movies['Titulo'].str.extract(r"\((\d{4})\)")

In [ ]:
movies.head()

id_filme                              Titulo  \
0         1                    Toy Story (1995)   
1         2                      Jumanji (1995)   
2         3             Grumpier Old Men (1995)   
3         4            Waiting to Exhale (1995)   
4         5  Father of the Bride Part II (1995)   

                                       Generos   Ano  
0  Adventure|Animation|Children|Comedy|Fantasy  1995  
1                   Adventure|Children|Fantasy  1995  
2                               Comedy|Romance  1995  
3                         Comedy|Drama|Romance  1995  
4                                       Comedy  1995

Como se pode ver no resultado acima, a coluna Ano foi criada

In [ ]:
#importante as colunas id_filme tem que ter o mesmo nome nas duas DF, se estiver movieId	para id_filme



---

**Questão 1 - b)** Qual o grupo mais disperso, o de usuários que avaliaram ou o de usuários que rotularam filmes?
<br/><br/>
Para responde isso vamos preparar as tabelas.




---
**Questão 1 - c)**  Qual a categoria rotulada mais frequente?
<br/><br/>




---

**Questão 1 - d)** Como estão distribuídas as avaliações ao longo do tempo?
<br/><br/>


**Questão 2 - Movies - a)** Qual o ano que mais houve lançamento de filmes?
<br/><br/>

In [26]:
anoMaiorNumLancamento = movies['Ano'].mode()
anoMaiorNumLancamento

0    2002
Name: Ano, dtype: object

Como podemos ver o ano que mais houve lançamentos foi em 2002.
<br/><br/>

---
<br/><br/>

**Questão 2 - Movies - b)** Existem quantos filmes de Aventura, quantos de Romance, e quantos com os dois juntos?

In [36]:
filtro = movies['Generos'].str.contains('Adventure')
filtro2 = movies['Generos'].str.contains('Romance')
filtro3 = movies['Generos'].str.contains(r'(?=.*Adventure)(?=.*Romance)', regex=True)

adventureMovies = movies[filtro]
romanceMovies = movies[filtro2]
adventureRomanceMovies = movies[filtro3]

In [28]:
movies[filtro]

id_filme                                 Titulo  \
0            1                       Toy Story (1995)   
1            2                         Jumanji (1995)   
7            8                    Tom and Huck (1995)   
9           10                       GoldenEye (1995)   
12          13                           Balto (1995)   
...        ...                                    ...   
9710    187595         Solo: A Star Wars Story (2018)   
9713    188301            Ant-Man and the Wasp (2018)   
9717    188833  The Man Who Killed Don Quixote (2018)   
9720    189333   Mission: Impossible - Fallout (2018)   
9731    191005                         Gintama (2017)   

                                          Generos   Ano  
0     Adventure|Animation|Children|Comedy|Fantasy  1995  
1                      Adventure|Children|Fantasy  1995  
7                              Adventure|Children  1995  
9                       Action|Adventure|Thriller  1995  
12                   Adventure|Animation|Children  1995  
...                                           ...   ...  
9710             Action|Adventure|Children|Sci-Fi  2018  
9713       Action|Adventure|Comedy|Fantasy|Sci-Fi  2018  
9717                     Adventure|Comedy|Fantasy  2018  
9720                    Action|Adventure|Thriller  2018  
9731               Action|Adventure|Comedy|Sci-Fi  2017  

[1263 rows x 4 columns]

In [32]:
movies[filtro2]

id_filme                               Titulo  \
2            3              Grumpier Old Men (1995)   
3            4             Waiting to Exhale (1995)   
6            7                       Sabrina (1995)   
10          11       American President, The (1995)   
14          15              Cutthroat Island (1995)   
...        ...                                  ...   
9639    179511                 Emerald Green (2016)   
9660    181315                Phantom Thread (2017)   
9691    184349                   Elsa & Fred (2005)   
9715    188751  Mamma Mia: Here We Go Again! (2018)   
9725    190207                          Tilt (2011)   

                              Generos   Ano  
2                      Comedy|Romance  1995  
3                Comedy|Drama|Romance  1995  
6                      Comedy|Romance  1995  
10               Comedy|Drama|Romance  1995  
14           Action|Adventure|Romance  1995  
...                               ...   ...  
9639  Adventure|Drama|Fantasy|Romance  2016  
9660                    Drama|Romance  2017  
9691             Comedy|Drama|Romance  2005  
9715                   Comedy|Romance  2018  
9725                    Drama|Romance  2011  

[1596 rows x 4 columns]

In [33]:

romanceMovies.shape

(1596, 4)

In [34]:
movies[filtro3]

id_filme                                Titulo  \
14          15               Cutthroat Island (1995)   
89         101                  Bottle Rocket (1996)   
222        258  Kid in King Arthur's Court, A (1995)   
320        362               Jungle Book, The (1994)   
337        380                      True Lies (1994)   
...        ...                                   ...   
8709    124851                       Delirium (2014)   
8713    125914                      Mortdecai (2015)   
8962    136859                     The Lovers (2015)   
9437    166635                     Passengers (2016)   
9639    179511                  Emerald Green (2016)   

                                        Generos   Ano  
14                     Action|Adventure|Romance  1995  
89               Adventure|Comedy|Crime|Romance  1996  
222   Adventure|Children|Comedy|Fantasy|Romance  1995  
320                  Adventure|Children|Romance  1994  
337    Action|Adventure|Comedy|Romance|Thriller  1994  
...                                         ...   ...  
8709                   Adventure|Romance|Sci-Fi  2014  
8713           Adventure|Comedy|Mystery|Romance  2015  
8962            Action|Adventure|Romance|Sci-Fi  2015  
9437             Adventure|Drama|Romance|Sci-Fi  2016  
9639            Adventure|Drama|Fantasy|Romance  2016  

[119 rows x 4 columns]

In [44]:
adventureQuant = adventureMovies.shape[0]
print('Filmes de aventura: ' + str(adventureQuant))

romanceQuant = romanceMovies.shape[0]
print('\nFilmes de romance: ' + str(romanceQuant))

adventureRomanceQuant = adventureRomanceMovies.shape[0]
print('\nFilmes de aventura e romance: ' + str(adventureRomanceQuant))

Filmes de aventura: 1263

Filmes de romance: 1596

Filmes de aventura e romance: 119


Criamos um filtro buscando gêneros específicos, e depois pegamos sua formação para ver suas linhas.
Assim temos:

*   Aventura: 1263 filmes.
*   Romance: 1596 filmes.
*   Aventura e Romance: 119 filmes.


---

<br/>

**Questão 2 - Links - a)** Escreva aqui a sua pergunta

resposta
<br/><br/>

---


**Questão 2 - Links - b)** Escreva aqui a sua pergunta



---

**Questão 2 - Tags - a)** Escreva aqui a sua pergunta

resposta
<br/><br/>

---

**Questão 2 - Tags - b)** Escreva aqui a sua pergunta

resposta



---



**Questão 2 - Ratings - a)** Escreva aqui a sua pergunta

resposta
<br/><br/>


---




**Questão 2 - Ratings - b)** Escreva aqui a sua pergunta





resposta



<br/><br/>
---

